### ISSUE TITLE
Support array types as BigQuery results

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#17045]
The Simba JDBC driver doesn't currently support BigQuery array types out of the box:

```java
try (ResultSet rs = connection.createStatement().executeQuery("select array[1, 2]")) {
    while (rs.next()) {
        println(rs.getArray(1));
    }
}
```

Raises this error:

```
10400
[Simba][JDBC](10400) Invalid type for data - column: 1, type: Array.
07006
java.sql.SQLDataException: [Simba][JDBC](10400) Invalid type for data - column: 1, type: Array.
	at com.simba.googlebigquery.exceptions.ExceptionConverter.toSQLException(Unknown Source)
	at com.simba.googlebigquery.jdbc.common.SForwardResultSet.getArray(Unknown Source)
	at org.jooq.testscripts.JDBC.main(JDBC.java:40)
```

But it is possible to fetch arrays in a serialised format. E.g.:

```java
try (ResultSet rs = connection.createStatement().executeQuery("select array[1, 2]")) {
    while (rs.next()) {
        println(rs.getObject(1));
    }
}
```

Prints:

```
{"v":[{"v":"1"},{"v":"2"}]}
```

The type of this `Object` is just `java.lang.String`, so we're on our own with the parsing, it seems.

**Comment by lukaseder** [comment]
This is how structs are serialised (which will be out of scope for now):

```java
try (ResultSet rs = connection.createStatement().executeQuery("select array[struct(1, 2), struct(2, 3)]")) {
    while (rs.next()) {
        println(rs.getObject(1));
    }
}
```

Resulting in:

```
{"v":[{"v":{"f":[{"v":"1"},{"v":"2"}]}},{"v":{"f":[{"v":"2"},{"v":"3"}]}}]}
```

**Comment by lukaseder** [comment]
Implemented in jOOQ 3.20.0
